buildpack/telemetry/appdynamics_telegraf_output.py

```diff
@@ -0,0 +1,181 @@
+        #     "timestamp": 1647939590,
+        #     }
+        # ]
+
+    """
+
+    fields = metric.get("fields")
+
+    converted = []
+
+    if fields is None:
+        logging.error(
+            "Converting metrics for AppDynamics Machine Agent: invalid format of specific metric (telegraf)."
+        )
+        return
+
+    for value_name in fields.keys():
+        value = fields[value_name]
+        if value_name == "value":
+            metric_name = metric["name"]
+        else:
+            metric_name = "_".join((metric["name"], value_name))
+
+        metric_path_list = [METRIC_TREE_BASE_NODE, metric_name]
+
+        tags = metric.get("tags")
+        if tags:
+            area = tags.get("area")
+            id = tags.get("id")
+            db = tags.get("db")
+            activity = tags.get("activity")
+            microflow = tags.get("microflow")
+
+            if area:
+                metric_path_list.append(area)
+            if id:
+                metric_path_list.append(id)
+            if db:
+                metric_path_list.append(f"Database '{db}'")
+            if microflow:
+                metric_path_list.append(microflow)
+            if activity:
+                metric_path_list.append(activity)
+
+        metric_path = "|".join(metric_path_list)
+
+        conv_metric = {
+            "metricName": metric_path,
+            "aggregatorType": AGGREGATION_TYPE,
+            "value": value,
+            "timestamp": metric["timestamp"],
+        }
+
+        converted.append(conv_metric)
+
+    return converted
+
+
+def convert_and_push_payload():
+    """
+    The function collect metrics json from STDIN (Telegraf 'output.exec')
+    and transform it to the structure of the compatible payload for the
+    AppDynamics Machine Agent HTTP listener.
+    AppDynamics Docs: https://docs.appdynamics.com/22.2/en/infrastructure-visibility/machine-agent/extensions-and-custom-metrics/machine-agent-http-listener
+
+    """
+
+    metrics_str = input()
+    metrics_dict = json.loads(metrics_str)
+    metrics_list = metrics_dict.get("metrics")
+
+    if metrics_list is None:
+        logging.error(
+            "Converting metrics for AppDynamics Machine Agent: invalid format of metrics json (telegraf)."
+        )
+        return
+
+    appdynamics_payload = []
+
+    for metric in metrics_list:
+        # Convert each metric from Telegraf json structure
+        # to the AppDynamics Machine Agent one.
+        converted_metrics = _convert_metric(metric)
+
+        if converted_metrics:
+            appdynamics_payload.extend(converted_metrics)
+
+    filtered_appd_payload = _filter_last(appdynamics_payload)
+
+    try:
+        resp = requests.post(
+            APPDYNAMICS_MACHINE_AGENT_URL, json=filtered_appd_payload
+        )
+        logging.info(
+            "Request to AppDynamics Machine Agent. Status code: {}.".format(
+                resp.status_code
+            )
+        )
+    except ConnectionError as e:
```

Comment:
Couple of things to check here. 
- Do we need to worry about any other exceptions other than `ConnectionError`?
- Should we add a timeout to the post request and catch `Timeout` exception?